pylhc/lsa_to_madx.py

```diff
@@ -0,0 +1,241 @@
+    """
+    Parses a file for LSA knobs and their trim values. Each line should be a knob name
+    following by a number of the trim value. If no value is written, it will be defaulted
+    to ``1.0``. Lines starting with a ``#`` are ignored.
+
+    Args:
+        knobs_file (~pathlib.Path): Path to the file with definitions.
+
+    Returns:
+        A `dict` with as keys the parsed knob names and as values their associated trims.
+    """
+    knob_lines = [line for line in Path(knobs_file).read_text().splitlines() if not line.startswith("#")]
+    results = {}
+
+    for line in knob_lines:
+        knob = line.split()[0]  # this is the knob name
+        try:  # catch the trim value, which is a number after the knob name
+            trim = float(line.split()[1])
+            results[knob] = trim
+        except IndexError:  # the is no number, we default the trim value to 1.0
+            LOG.debug(f"No trim value was specified for knob '{knob}' - defaulting to 1.0")
+            results[knob] = 1.0
+    return results
+
+
+def get_madx_script_from_definition_dataframe(deltas_df: tfs.TfsDataFrame, lsa_knob: str, trim: float = 1.0) -> str:
+    """
+    Given the extracted definition dataframe of an LSA knob - as returned by
+    `~pylhc.data_extract.lsa.LSAClient.get_knob_circuits` - this function will generate the
+    corresponding ``MAD-X`` text commands that would reproduce this knob in a script.
+
+    Args:
+        deltas_df (~tfs.frame.TfsDataFrame): The extracted definition dataframe of an ``LSA`` knob. This
+            can be obtained with `~pylhc.data_extract.lsa.LSAClient.get_knob_circuits`.
+        lsa_knob (str): The complete ``LSA`` name of the knob, including any ``LHCBEAM/`` etc.
+        trim (float): The trim value to write for the knob. Defaults to 1.
+
+    Returns:
+        The complete ``MAD-X`` script to reproduce the knob, as a string.
+    """
+    LOG.debug(f"Determining MAD-X commands to reproduce knob '{lsa_knob}'")
+    change_commands = [f"! Start of change commands for knob: {lsa_knob}"]
+
+    # Set this to 1 by default but can be changed by the user to reproduce a given trim
+    knob_itself = lsa_knob.split("/")[1]  # without the LHCBEAM/ part
+    trim_variable = f"{knob_itself}_trim"
+    change_commands.append("! Change this value to reproduce a different trim")
+    change_commands.append(f"! Beware some knobs are not so linear in their trims")
+    change_commands.append(f"{trim_variable} = {trim};")
+    change_commands.append("! Impacted variables")
+
+    for variable, delta_k in deltas_df.DELTA_K.items():
+        change_commands.append(f"{variable:<12} = {variable:^15} + {delta_k:^25} * {trim_variable};")
+    change_commands.append(f"! End of change commands for knob: {lsa_knob}\n")
+    return "\n".join(change_commands)
+
+
+# ----- Script Part ----- #
+
+
+def _get_args():
+    """Parse Commandline Arguments."""
+    parser = argparse.ArgumentParser(
```

Comment:
why argparse and generic parser? with generic parser you could make it a fct and combine extract knobs and this

Reply:
I'll keep this script simple for now, if we want to do this in the future we can always change the parser without breaking functionality